crowbar_framework/app/models/api/node.rb

```diff
@@ -257,9 +257,27 @@ def repocheck(options = {})
                                                                    architecture)
                   ::Openstack::Upgrade.enable_repos_for_feature(feature, Rails.logger)
                 end
-                available, repolist = ::Crowbar::Repository.provided_and_enabled_with_repolist(
+                _, repolist = ::Crowbar::Repository.provided_and_enabled_with_repolist(
                   feature, platform, architecture
                 )
+                # Filter the result and remove the repositories that are missing but not mandatory
+                repolist.each do |category, repomap|
+                  if [:missing, :inactive].include? category
+                    repomap.each do |arch, repos|
+                      repos.reject! { |repo| repolist[:optional][arch].include?(repo) }
```

Comment:
1. might this be `repos -= repolist[:optional][arch]` ?
2. looks like `repolist[:optional]` might not be defined if all repos under given feature are "mandatory"

Reply:
for #2 we need something like `next unless repolist.include? :optional` before the reject! line

Reply:
Ah true. Or I can make the lib function always return at least empty list for optional